paddle/fluid/eager/auto_code_generator/generator/eager_gen.py

```diff
@@ -1826,16 +1833,19 @@ def GenerateHigherOrderNodeCreationCode(self):
         is_invoke_forward_api = IsInvokeForwardApi(
             self.grad_api_contents, self.forward_apis_dict
         )
+        is_composite_forward_api = False if self.composite_func_info == [] else True
+        
         if next_node_generator is not None:
             has_higher_order_node = True
             return (
                 has_higher_order_node,
                 is_invoke_forward_api,
+                is_composite_forward_api,
                 next_grad_node_creation_str,
                 next_grad_node_out_list,
                 next_node_generator.backward_forward_inputs_map,
             )
-        elif not is_invoke_forward_api:
+        elif not is_invoke_forward_api and not is_composite_forward_api:
```

Comment:
what if it hits else branch?

Reply:
if is_invoke_farward_api or is_composite_grad_api next_grad_node_creation_str should be none, we will add this when delete Flags_prim_enabled
add TODO here